File: 9041/lab07/log_probability.py
#!/usr/bin/python3

import sys
import re
import glob
import math


target = sys.argv[1]
names = []
occur_time = {}
total_words = {}


for file in glob.glob("lyrics/*.txt"):
    occur = 0
    total = 0
    file_name_1 = re.split('.txt', file)
    file_name_2 = re.split('/', file_name_1[0])
    name = file_name_2[1]
    name = re.sub('_', ' ', name)
    names.append(name)
    with open(file) as F:
        for line in F:
            for char in re.split('[^a-zA-Z]', line):
                if (char.lower() == target.lower()):
                    occur += 1
                if (char != ""):
                    total += 1
    occur_time[name] = occur 
    total_words[name] = total

for key in sorted(occur_time):
   print("log((%d+1)/%6d) = %8.4f %s" % (occur_time[key], total_words[key],
           math.log((occur_time[key]+ 1)/total_words[key]),  key))
    

File: 9041/lab07/log_probability.pl
#!/usr/bin/perl -w




foreach $file (glob "lyrics/*.txt") {
	
	@file_name_1 = split /.txt/, $file;
	@file_name_2 = split /\//, $file_name_1[0]; 
	$file_name_2[1] =~ s/_/ /g;
	$name = $file_name_2[1];
	
	open my $f, '<', $file or die;
	while ($line = <$f>) {
	chomp $line;
 	@content = split /[^a-zA-Z]/,$line;
		foreach $str (@content){
			foreach $word (@ARGV) {
				if (lc $str eq lc $word) {
		   			push @result, $str;		
				}
			}
			
			if ($str ne "") {
		   		push @result2, $str;		
			}
		}
	}
	$result_count{$name} = $#result + 1;
	$result_word{$name} = $#result2 + 1;
	@result = ();
	@result2 = ();

}
foreach my $key (sort keys %result_count) {
	$fren = log(($result_count{$key} + 1) / $result_word{$key});
	printf "log((%d+1)/%6d) = %8.4f %s\n", ($result_count{$key}, $result_word{$key}, $fren, $key);

}
